Px.Utils/Validation/SyntaxValidation/SyntaxValidationUtilityMethods.cs

```diff
@@ -0,0 +1,267 @@
+        public static bool IsDateTimeFormat(string input, char stringDelimeter, string dateTimeFormat)
+        {
+            if (!IsStringFormat(input, stringDelimeter))
+            {
+                return false;
+            }
+            return DateTime.TryParseExact(input.Trim(stringDelimeter), dateTimeFormat, CultureInfo.InvariantCulture, DateTimeStyles.None, out _);
+        }
+
+        /// <summary>
+        /// Determines whether a string is in a string format.
+        /// </summary>
+        /// <param name="input">The input string to check</param>
+        /// <param name="stringDelimeter">The delimeter that encloses the string</param>
+        /// <returns>Returns a boolean which is true if the input string is in a string format</returns>
+        public static bool IsStringFormat(string input, char stringDelimeter)
+        {
+            return input.StartsWith(stringDelimeter) && input.EndsWith(stringDelimeter);
+        }
+
+        /// <summary>
+        /// Determines whether a string is in a list format.
+        /// </summary>
+        /// <param name="input">The input string to check</param>
+        /// <param name="listDelimeter">The delimeter that separates the list items</param>
+        /// <param name="stringDelimeter">The delimeter that encloses the string</param>
+        /// <returns>Returns a boolean which is true if the input string is in a list format</returns>
+        public static bool IsStringListFormat(string input, char listDelimeter, char stringDelimeter)
+        {
+            string[] list = input.Split(listDelimeter);
+            if (list.Length <= 1)
+            {
+                return false;
+            }
+            else if (Array.TrueForAll(list, x => IsStringFormat(x.Trim(), stringDelimeter)))
+            {                
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Determines whether a string is in a boolean format.
+        /// </summary>
+        /// <param name="input">The input string to check</param>
+        /// <param name="booleanYes">The string that represents a boolean true value</param>
+        /// <param name="booleanNo">The string that represents a boolean false value</param>
+        /// <returns>Returns a boolean which is true if the input string is in a boolean format</returns>
+        public static bool IsBooleanFormat(string input, string booleanYes, string booleanNo)
+        {
+            return input == booleanYes || input == booleanNo;
+        }
+
+        /// <summary>
+        /// Determines whether a string is in a number format.
+        /// </summary>
+        /// <param name="input">The input string to check</param>
+        /// <returns>Returns a boolean which is true if the input string is in a number format</returns>
+        public static bool IsNumberFormat(string input)
```

Comment:
I suggest using regexp. That way the wanted format is easier to understand when reading the code.